src/components/date/date.spec.tsx

```diff
@@ -739,13 +856,15 @@ describe("Date", () => {
         it.each(validValues)(
           "it updates the input value from %s to the expected format on blur",
           (value) => {
-            wrapper.find("input").simulate("change", { target: { value } });
-            expect(wrapper.update().find("input").prop("value")).toEqual(value);
+            wrapper?.find("input").simulate("change", { target: { value } });
+            expect(wrapper?.update().find("input").prop("value")).toEqual(
+              value
+            );
             simulateBlurOnInput(wrapper);
-            const { separator } = locales[localeKey];
+            const { separator } = locales[localeKey as keyof typeof locales];
```

Comment:
**suggestion** (non-blocking): It won't let me comment on line 832, but similarly to changes suggested above, if that is changed to `describe.each(Object.keys(locales) as (keyof typeof locales)[])(` then not only do I think that is more easy to tell the assertion is sensible, it enables it to be removed from both here and line 886